codalab/lib/bundle_cli.py

```diff
@@ -2428,6 +2434,8 @@ def follow_targets(self, client, bundle_uuid, subpaths, from_start=False):
                 # file and if so, initialize the offset.
                 if subpath_is_file[i] is None:
                     target_info = client.fetch_contents_info(bundle_uuid, subpaths[i], 0)
+                    bundle_uuid = info.get('resolved_uuid', bundle_uuid)
+                    subpath = info.get('resolved_path', subpath)
```

Comment:
Is there a way to avoid the proliferation of these redirections on the client side?  Maybe these should be wrapped or pushed into `fetch_contents_info`?  Not sure if that's the right call, but just bringing it up.

Reply:
Despite all the code lines there's not really much "logic" on the client side here. It would be more aesthetically pleasing to maybe have all the other `contents`-related methods take a unified `target_info` object so the client doesn't have to do any variable extraction from the results of `get_target_info` and just passes its return value directly downstream, or maybe all other contents getting methods may do the `get_target_info` call server-side to double check. I'll look into the viability of either.